test/core/plaid_link_test.dart

```diff
@@ -0,0 +1,149 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mocktail/mocktail.dart';
+import 'package:plaid_flutter/plaid_flutter.dart';
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+
+class _MockPlaidPlatformInterface extends Mock
+    with MockPlatformInterfaceMixin
+    implements PlaidPlatformInterface {}
+
+class _MockLinkEvent extends Mock implements LinkEvent {}
+
+class _MockLinkExit extends Mock implements LinkExit {}
+
+class _MockLinkSuccess extends Mock implements LinkSuccess {}
+
+main() {
+  group('PlaidLink', () {
+    late PlaidPlatformInterface plaidPlatformInterface;
+
+    setUp(() {
+      plaidPlatformInterface = _MockPlaidPlatformInterface();
+      PlaidPlatformInterface.instance = plaidPlatformInterface;
+    });
+
+    test('can be instantiated', () {
+      var plaidLink = PlaidLink();
+      expect(plaidLink, isNotNull);
+    });
+
+    group('onEvent', () {
+      late LinkEvent linkEvent;
+
+      setUp(() {
+        linkEvent = _MockLinkEvent();
+        when(
+          () => plaidPlatformInterface.onObject,
+        ).thenAnswer((_) => Stream.fromIterable([linkEvent]));
+      });
+
+      test('emits a LinkEvent', () async {
+        await expectLater(PlaidLink.onEvent, emits(linkEvent));
+      });
+
+      test('calls platform.onObject', () {
+        PlaidLink.onEvent;
+
+        verify(
+          () => plaidPlatformInterface.onObject,
+        ).called(1);
+      });
+    });
+
+    group('onExit', () {
+      late LinkExit linkExitEvent;
+
+      setUp(() {
+        linkExitEvent = _MockLinkExit();
+        when(
+          () => plaidPlatformInterface.onObject,
+        ).thenAnswer((_) => Stream.fromIterable([linkExitEvent]));
+      });
+
+      test('emits a LinkExit', () async {
+        await expectLater(PlaidLink.onExit, emits(linkExitEvent));
+      });
+
+      test('calls platform.onObject', () {
+        PlaidLink.onExit;
+
+        verify(
+          () => plaidPlatformInterface.onObject,
+        ).called(1);
+      });
+    });
+
+    group('onSuccess', () {
+      late LinkSuccess linkSuccess;
+
+      setUp(() {
+        linkSuccess = _MockLinkSuccess();
+        when(
+          () => plaidPlatformInterface.onObject,
+        ).thenAnswer((_) => Stream.fromIterable([linkSuccess]));
+      });
+
+      test('emits a LinkSuccess', () async {
+        await expectLater(PlaidLink.onSuccess, emits(linkSuccess));
+      });
+
+      test('calls platform.onObject', () {
+        PlaidLink.onSuccess;
+
+        verify(
+          () => plaidPlatformInterface.onObject,
+        ).called(1);
+      });
+    });
+
+    group('open', () {
+      test('calls platform.open', () async {
+        const token = 'token';
+
+        final linkTokenConfiguration = LinkTokenConfiguration(token: token);
```

Comment:
I'm having a hard time to test the `LinkTokenConfiguration` class since it doesn't override the `==` operator as well as `hashCode` for comparison.

Anyways, I can create a fake for it but  makes the test harder to read.
I can submit a PR for this if the author wants.

cc @MauriMiguez @jhionan

Reply:
Here's my current code to test if this `PlaidLink` sdk was successfully being called within my repository.
Feel free to suggest for improvements.
```
group('openPlaidLink', () {
      test('opens the plaid sdk successfully', () async {
        when(() => apiClient.requestPlaidLinkToken(email: email)).thenAnswer(
          (_) => Future.value(plaidTokenResponse),
        );
        await plaidRepository.requestPlaidToken(email: email);

        when(
          () => plaidPlatformInterface.open(
            configuration: any(
              named: 'configuration',
              that: isA<LinkTokenConfiguration>().having(
                (l) => l.token,
                'token',
                plaidToken.token,
              ),
            ),
          ),
        ).thenAnswer((_) => Future.value());
        await plaidRepository.openPlaidLink();

        verify(
          () => plaidPlatformInterface.open(
            configuration: any(
              named: 'configuration',
              that: isA<LinkTokenConfiguration>().having(
                (l) => l.token,
                'token',
                plaidToken.token,
              ),
            ),
          ),
        ).called(1);
      });
    });
```

Reply:
My 1 cent contribution:

I see no harm by adding equality inside `LinkTokenConfiguration` it don't have any implication in the code, but would make the tests cleaner

Reply:
Generally I don't prefer the idea of changing the code because of tests. However, in this particular case, I don't there is too much harm on adding equatable to `LinkTokenConfiguration` so It makes testing easier. 

In this particular code I wrote to test the PlaidLink class I didn't have trouble with that since I did use the same instance to mock and to verify the `open` method. However in my project I did have to do the same as @fernan542 to accomplish my unit tests.

I can also add it no problem but I don't think it's my call. Are you interested in adding that @jorgefspereira ?

Reply:
@MauriMiguez Thank you for the tests, its really helpful :) go ahead and add equality to the LinkTokenConfiguration object. I don't see any issue in adding that and makes everything more clean. Added it and then I'll merge the pull request.

Reply:
Just added Equatable to the project and made LinkTokenConfiguration extend it. 

Note: I had to make receivedRedirectUri final as Equatable is inmutable. And from the docs `A class is immutable if all of the instance fields of the class, whether defined directly or inherited, are final.`